xdsl/ir/core.py

```diff
@@ -372,6 +372,13 @@ def erase(self, safe_erase: bool = True) -> None:
             )
         self.replace_by(ErasedSSAValue(self.type, self))
 
+    def __hash__(self):
+        """
+        Make SSAValue hashable. Two SSA Values are never the same, therefore
+        the use of `id` is allowed here.
+        """
+        return id(self)
```

Comment:
It is a bit weird to have this in an ABC, right? Should we have an assertion here that this is not executed?

Reply:
Hm, yeah, it's kind of redundant there now....